sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -129,7 +129,7 @@ object TableOutputResolver {
     }
   }
 
-  private def checkNullability(
+  private[analysis] def checkNullability(
```

Comment:
I want to reuse code from `TableOutputResolver` wherever possible but adding assignment processing would make it even more complicated than it is today. That's why I decided to open up some methods instead.

Let me know your thoughts.